src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -666,7 +666,7 @@ protected virtual void RequestRun()
         /// </summary>
         protected virtual void NodeModified(NodeModel node)
         {
-
+            HasUnsavedChanges = true;
```

Comment:
Homeworkspace and CustomWorkspace - uses base.NodeModified(node). That is the reason why I have used this on the base class. I can say HasUnsavedChanges = true on PortModel. I thought adding here makes more sense.